app/controllers/api/vms_controller.rb

```diff
@@ -59,6 +59,14 @@ def suspend_resource(type, id = nil, _data = nil)
     end
     central_admin :suspend_resource, :suspend
 
+    def parent_choices_resource(ids)
+      parent_choices = {}
+      parent_choices[Digest::MD5.hexdigest(ids.inspect)] ||= begin
```

Comment:
you just created `parent_choices`. So the `||=` doesn't make sense to me.

According to the structure of this code, I was expecting to see a loop on `ids` and populating multiple `parent_choices`.

But I'm not totally clear what you are doing here, so don't take that as an endorsement of one way or another, just saying what it looked like you were trying to do.

Or I was expecting to see `@parent_choices ||= {}`
again, just trying to guess at your intentions